Automattic/sensei
# Add error handlers

### Changes proposed in this Pull Request

* Add error handlers to issues mentioned [here](https://github.com/Automattic/sensei/pull/3507), and show them friendly.

[Exporter issue](https://github.com/Automattic/sensei/pull/3507#issuecomment-675603526):
* For now, it handles generic errors in the exporter.
* For now, if there is more than one, I just concatenated it in the same message (See screenshot).
* It's using the data port logs, but as now we're just adding generic errors, it's not related to any line.

[Importer issue](https://github.com/Automattic/sensei/pull/3507#issuecomment-677542869):
* It also handles an importer issue, when the imported file doesn't exist (It happens in the sample import if the file doesn't exist).

### Testing instructions

Exporter issue:
* Replace the line `$output_file = new SplFileObject( $this->file, 'a' );` (In `Sensei_Export_Task::run`) to `$output_file = new SplFileObject( '', 'a' );`
* Go to the exporter, select content types, and Continue.
* Make sure a friendly error is showed after the exporting, and the `debug.log` doesn't contain errors.

Importer issue:
* Merge the `add/sample-course` here, as a way to simulate the issue.
* Remove or rename the `/sample-data` folder.
* Go to the Setup Wizard.
* Go to the ready step.
* Import the sample data. Make sure it shows a friendly error and any error appears in the `debug.log`.


### Screenshot / Video

<img width="767" alt="Screen Shot 2020-08-20 at 19 00 12" src="https://user-images.githubusercontent.com/876340/90830191-9f9f6d80-e317-11ea-9df4-6967e969c460.png">
